### PR TITLE
Add Interlacing

### PR DESCRIPTION
**Pull request by Teranode**
This adds interlacing to png/gif/jpeg (webp excluded cause idk if it uses it)

### CODE REVIEW

**Comment by LukeTowers** [comment]
Could you open a PR to the docs repo documenting this new option please?

**Comment by Teranode** [comment]
Here you go:
https://github.com/octobercms/docs/pull/244

**Comment by daftspunk** [comment]
Related issues:

https://github.com/octobercms/library/pull/267
https://github.com/octobercms/docs/pull/244
https://github.com/octobercms/docs/pull/245
https://github.com/octobercms/october/issues/2795

**Comment by daftspunk** [comment]
Any reason why this shouldn't be enabled by default? Seems like all upside with no downside, right?

File: src/Database/Attach/Resizer.php [hunk 1]
```diff
@@ -125,16 +125,18 @@ public function reset()
...
     {
         $this->options = array_merge([
-            'mode'    => 'auto',
-            'offset'  => [0, 0],
-            'sharpen' => 0,
-            'quality' => 90
+            'mode'      => 'auto',
+            'offset'    => [0, 0],
+            'sharpen'   => 0,
+            'interlace' => 0,
```
Comment (daftspunk) [inline src/Database/Attach/Resizer.php]: If this is a boolean it should probably be defined as such.

**Comment by LukeTowers** [comment]
@daftspunk @Teranode  It only really makes sense to enable it by default on JPEGs, and only if `imageinterlace()` actually creates a progressive JPEG instead of an interlaced one. See https://stackoverflow.com/questions/13449314/when-to-interlace-an-image for more information

**Comment by daftspunk** [comment]
Let's factor this in to the code then, interlace default true for JPG, false for all others.

**Comment by Teranode** [comment]
I usually have it disabled by default because the images are slightly larger that non-progressives

**Comment by daftspunk** [comment]
According to the answer provided it makes JPG files smaller...

**Comment by Teranode** [comment]
Im not sure if that answer is valid, the output differs case-by-case

**Comment by daftspunk** [comment]
Fair enough. Why are we calling `imageinterlace()` at all if the value is set to false?

**Comment by LukeTowers** [comment]
Also we shouldn't be treating image types differently if there is only one option for all three

**Comment by LukeTowers** [comment]
@Teranode don't forget to update the docs please :)

**Comment by Teranode** [comment]
@LukeTowers i was factoring in the fact that I had submit a PR for WEBP at the time, i wasn't sure if it used interlacing so I had the options separated.

**Comment by Teranode** [comment]
@LukeTowers @daftspunk would that work?

**Comment by LukeTowers** [comment]
WEBP doesn't use interlacing AFAIK. If you aren't using `$interface` more than once, then you should just make the call to `getOption()` within the conditional.

**Comment by daftspunk** [comment]
Sorry to be a pain, but please add a space in between if and the bracket.

```
if(
```

Should be

```
if (
```

**Pull request merged**